nix-darwin/darwin-configuration.nix

```diff
@@ -0,0 +1,72 @@
+{ config, pkgs, ... }:
+
+let
+  opsLib = import <iohk-ops/lib.nix>;
+
+in {
+  imports = [ ];
+
+  # List packages installed in system profile. To search by name, run:
+  # $ nix-env -qaP | grep wget
+  environment.systemPackages = with pkgs; [
+    nix-repl
+    nix
+    tmux
+    ncdu
+    git
+  ] ++ (if pkgs.stdenv.isDarwin then [
+    darwin.cctools
+  ] else []);
+
+  # Create /etc/bashrc that loads the nix-darwin environment.
+  programs.bash.enable = true;
+  # programs.zsh.enable = true;
+  # programs.fish.enable = true;
+
+  # Used for backwards compatibility, please read the changelog before changing.
+  # $ darwin-rebuild changelog
+  system.stateVersion = 2;
+
+  # You should generally set this to the total number of logical cores in your system.
+  # $ sysctl -n hw.ncpu
+  nix.maxJobs = 4;
+  nix.buildCores = 1;
+  nix.useSandbox = false;  # this seems to break things when enabled
+  nix.extraOptions = ''
+    gc-keep-derivations = true
+    gc-keep-outputs = true
+  '';
+
+  nix.binaryCachePublicKeys = [ "hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=" ];
+  nix.binaryCaches = [ "https://hydra.iohk.io" ];
+  nix.trustedUsers = [ "@admin" ];
+
+  nix.nixPath = [
+    "nixpkgs=https://nixos.org/channels/nixpkgs-unstable/nixexprs.tar.xz"
```

Comment:
would be nice if we could pin this to `unstablePkgs` commit as well

Reply:
Maybe this is a bit confusing and magic... the values in this file actually get sedded by the deploy script. I could remove these initial values to make it clearer?

The deploy script pins to the 17.09 channel but not a specific commit. Is that ok?

Reply:
if it's 17.09 then we can just use `fetch-nixpkgs.nix` in the root folder :)

Do we need to sed it? Can't we just put those here?

Reply:
The nix path is defined in haskell because it's needed before running the nix-darwin installer. I have edited the nix to make it clear that the values are templated.